public/app/features/alerting/unified/components/rules/central-state-history/EventDetails.tsx

```diff
@@ -0,0 +1,239 @@
+      </Text>
+    );
+  }
+
+  const getTransitionsCountByRuleUID = (ruleUID: string) => {
+    return logRecords.filter((record) => record.line.ruleUID === ruleUID).length;
+  };
+
+  return (
+    <Stack direction="column" gap={0.5}>
+      <Stack direction={'row'} gap={6}>
+        <StateTransition record={record} />
+        <ValueInTransition record={record} />
+        <NumberTransitions transitions={ruleUID ? getTransitionsCountByRuleUID(ruleUID) : 0} />
+      </Stack>
+      <Annotations rule={rule} />
+      <QueryVizualization rule={rule} ruleUID={ruleUID} logRecords={logRecords} />
+    </Stack>
+  );
+}
+
+interface StateTransitionProps {
+  record: LogRecord;
+}
+function StateTransition({ record }: StateTransitionProps) {
+  return (
+    <Stack gap={0.5} direction={'column'}>
+      <Text variant="body" weight="light" color="secondary">
+        <Trans i18nKey="central-alert-history.details.state-transitions">State transition</Trans>
+      </Text>
+      <Stack gap={0.5} direction={'row'} alignItems="center">
+        <EventState state={record.line.previous} showLabel />
+        <Icon name="arrow-right" size="lg" />
+        <EventState state={record.line.current} showLabel />
+      </Stack>
+    </Stack>
+  );
+}
+
+interface AnnotationsProps {
+  rule: CombinedRule;
+}
+const Annotations = ({ rule }: AnnotationsProps) => {
+  const styles = useStyles2(getStyles);
+  const annotations = rule.annotations;
+  if (!annotations) {
+    return null;
+  }
+  return (
+    <>
+      <Text variant="body" color="secondary" weight="light">
+        <Trans i18nKey="central-alert-history.details.annotations">
+          Annotations
+        </Trans>
+      </Text>
+      {Object.keys(annotations).length === 0 ? (
+        <Text variant="body" weight="light" italic>
+          <Trans i18nKey="central-alert-history.details.no-annotations">
+            No annotations
+          </Trans>
+        </Text>
+      ) : (
+        <div className={styles.metadataWrapper}>
+          {Object.entries(annotations).map(([name, value]) => (
+            <MetaText direction="column" key={name}>
+              {name}
+              <AnnotationValue value={value} />
+            </MetaText>
+          ))}
+        </div>
+      )}
+    </>
+  );
+};
+
+/**
+ *
+ * This component renders the visualization for the rule condition values over the selected time range.
+ * The visualization is a time series graph with the condition values on the y-axis and time on the x-axis.
+ * The values are extracted from the log records already fetched from the history api.
+ * The graph is rendered only if the rule is a Grafana rule.
+ *
+ */
+interface QueryVizualizationProps {
+  ruleUID: string;
+  rule: CombinedRule;
+  logRecords: LogRecord[];
+}
+const QueryVizualization = ({ ruleUID, rule, logRecords }: QueryVizualizationProps) => {
+  if (!isGrafanaRulerRule(rule?.rulerRule)) {
+    return (
```

Comment:
this is not possible, but we hace to check it

Reply:
As you only use the rulerRule here, could you pass that in instead, and perform the check in the parent component?

Then the prop could be `rulerRule: RulerGrafanaRuleDTO`?